### PR TITLE
Turn on frequencyio only on CIRCUITPY_FULL_BUILD

### PR DESCRIPTION
**Pull request by dhalbert**
`frequencyio` is too large to fit on some small builds due to language string size. Turn it off for small builds.

### CODE REVIEW

**Comment by sommersoft** [comment]
@dhalbert just to note, I went with explicit `CIRCUITPY_FREQUENCYIO = 0` so that SAMD21 wouldn't include it (I did miss setting it in CatWAN). I still need to go back and test the SAMD21 code that is left in `FrequencyIn` as an optional build. But, if a future SAMD21 port doesn't explicitly exclude `frequencyio`, and uses the full build, it will be included.

**Comment by dhalbert** [comment]
I can turn it off explicitly in all SAMD21 builds now. I can add a new commit to this PR. Does that make sense?

**Comment by dhalbert** [comment]
I would make the changes in mpconfigport.mk and check for SAMD21.

**Comment by sommersoft** [comment]
> I would make the changes in mpconfigport.mk and check for SAMD21.

why didn't I think of that. haha

**Comment by dhalbert** [comment]
`CIRCUITPY_FREQUENCYIO = 0` is left in on on samd21 `mpconfigboard.mk` files because even those builds need it off no matter, for space reasons, even it becomes usable on samd21.

File: ports/atmel-samd/mpconfigport.mk [hunk 1]
```diff
@@ -14,6 +14,12 @@ ifeq ($(LONGINT_IMPL),LONGLONG)
 MPY_TOOL_LONGINT_IMPL = -mlongint-impl=longlong
 endif
 
+# Put samd21-only choices here.
+ifeq ($(CHIP_FAMILY),samd51)
```
Comment (sommersoft) [inline ports/atmel-samd/mpconfigport.mk]: needs to be `samd21`?

Comment (dhalbert) [reply ports/atmel-samd/mpconfigport.mk]: Yes!

Comment (dhalbert) [reply ports/atmel-samd/mpconfigport.mk]: Thanks!

**Comment by dhalbert** [comment]
@sommersoft @tannewt this has now passed travis too.

**Pull request merged**